zerver/views/messages.py

```diff
@@ -689,8 +691,7 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
         query = after_query
     else:
         # This can happen when a narrow is specified.
-        query = query.where(inner_msg_id_col == anchor)
-
+        query = query.where(and_(inner_msg_id_col == anchor, inner_msg_id_col >= first_visible_message_id))
     main_query = alias(query)
     query = select(main_query.c, None, main_query).order_by(column("message_id").asc())
```

Comment:
I feel like the three codepaths above are kind of inconsistent, because sometimes we use max() and sometimes we use and_.  Does it make more sense to just have one place where we say:

~~~ py
if first_visible_message_id:
    query = query.where(inner_msg_id_col >= first_visible_message_id
~~~

This will occasionally translate to silly SQL like `...where message_id >= 40 and message_id >= 2342365`, but I would predict with near certainty that postgres does the right thing when you have redundant `>=` checks.